CMakeLists.txt

```diff
@@ -272,6 +276,9 @@ option(QUDA_DOWNLOAD_USQCD "Download USQCD software as requested by QUDA_QMP / Q
 option(QUDA_DOWNLOAD_ARPACK "Download ARPACK-NG software as requested by QUDA_ARPACK" OFF)
 option(QUDA_DOWNLOAD_OPENBLAS "Download OpenBLAS software as requested by QUDA_OPENBLAS" OFF)
 option(QUDA_JITIFY "build QUDA using Jitify" OFF)
+option(QUDA_DOWNLOAD_NVSHMEM "Download NVSHMEM" OFF)
+set(QUDA_DOWNLOAD_NVSHMEM_TAR "" CACHE STRING "location of NVSHMEM tarball")
```

Comment:
What's your reason for passing in a location of the `NVSHMEM` tarball as opposed to having QUDA automatically download a fixed one, similar to what's done with QMP/QIO? I can see argument in either direction, so I'm just curious.